### ISSUE TITLE
implement external networks connections

### ISSUE THREAD

**Issue by watchoutfreedom** [helpbuttons/hb-back#21]
We need to think how to link external helpbuttons projects to federate. till now The main platform could show links in the maps to external HB projects and redirect to see them.

**Comment by albjeremias** [comment]
Implement activityPub on the backend?!

https://hacks.mozilla.org/2018/11/decentralizing-social-interactions-with-activitypub/

**Comment by watchoutfreedom** [comment]
@albjeremias I think activity pub looks perfect for lots of the stuff we want to do. Do you think we can sync different button templates with it? I guess it would be an optional feature that allows the user to connect networks on other servers ? That would be amazing

**Comment by albjeremias** [comment]
it can be the "privacy" set of the network...
but i think the privacy of the network should be the default for a new button on that network, but each button could then choose:
 - Federated
 - Only this server
 - Only this network

**Comment by watchoutfreedom** [comment]
i think it can be depending on the settings of the network and not customizable by the user for the button. If you add the button to a federated network you assume it would be shared with the federation, and the same with a private one.

**Comment by albjeremias** [comment]
what about having both? 
I mean on the Network they can have another option like: "enforce privacy settings" and so that the buttons wouldn't be able to change it.
but the question is.. what happens when the button is in many different networks?

**Comment by watchoutfreedom** [comment]
if it's in different networks you assume the less restrictive criteria is going to apply. 

 I don't think it's needed to allow this per button but if you think it's I'm open to hear main reasons for it.

**Comment by albjeremias** [comment]
> if it's in different networks you assume the less restrictive criteria is going to apply.
i dont think that this is a good approach, can be very deceive for a user that creates a button and suddenly the button becomes what the owner of the network defines. Privacy needs to be the main focus in any webapp. Also I don't think that it's an easy solution to check all the networks of a button and see which on is less restrictive.

> I don't think it's needed to allow this per button but if you think it's I'm open to hear main reasons for it.
I'd rather hear the main reasons why not! ;)

> Also multiple conflicts between button settings and net settings can lead to confusing cases. You can also deactivate your button or remove from any net if don't want it to be present already.
It's important that this is specified.. and get clear about this conflicts.

**Comment by watchoutfreedom** [comment]
Ok. Let me put together privacy setttings beacause now i realized we were missing fields in the data model . So we can define an enum in the model like this :

Users: 
ButtonsPrivacy : 
-active-private  (only visible by the users of the network it's added)
-active-public (visible by any user of the same server)
-active-federated  (visible by the network and any other federated network)


Buttons 
ButtonPrivacy:
-unactive ( invisible for other users, admins can still see it)
-active  (tells if the the buttons is visible for other users (default privacy set by user))
-active-private  (only visible by the users of the network it's added)
-active-public (visible by any user of the same server)
-active-federated  (visible by the server and any other federated network)

**Comment by albjeremias** [comment]
> Ok. Let me put together privacy setttings beacause now i realized we were missing fields in the data model . So we can define an enum in the model like this :
> 
> Users:
> ButtonsPrivacy :
> -active-private (only visible by the users of the network it's added)
> -active-public (visible by any user of the same server)
> -active-federated (visible by the network and any other federated network)
> 
> Buttons
> ButtonPrivacy:
> -unactive ( invisible for other users, admins can still see it)
> -active (tells if the the buttons is visible for other users (default privacy set by user))
> -active-private (only visible by the users of the network it's added)
> -active-public (visible by any user of the same server)
> -active-federated (visible by the server and any other federated network)

Does this mean that its not by network.. but by Button & ButtonUser ?!
I can understand, but it's very confusing.. if a user picks a Button it already knowns that it will have its privacy setting... maybe with a use case we can explore it better to understand.

**Comment by watchoutfreedom** [comment]
This is what  understood from your proposal sorry. I prefer it to bbe handled by network but I understood you want it by button and user / XD

**Comment by watchoutfreedom** [comment]
You can see an example (very rough styling yet) of the button page with the feed underneath. The idea is to have a feed in which you can see on messages and las updates on that button.  That feed can show/hide parts depending on the user that looks to it. for instance, if an user starts a conversation / send a message / interacts and the button channel is set to indiividual, only the owner of the button and the writer can see that particular feed. 

can be seen here

http://localhost:3000/ButtonFile?
